### PR TITLE
Add coderabbit config and clean up configuration

### PR DESCRIPTION
**Pull request by adamwg**
### Description of your changes

Three small changes to our repository configuration:

1. Add a coderabbit config based on Crossplane's. We'll get coderabbit running on this repository once this is merged.
2. Add @jcogilvie and @tampakrap to the CODEOWNERS file.
3. Stop linting protobufs and pushing them to the Buf schema registry. Our protos are vendored from crossplane/crossplane where they are linted and pushed.

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `./nix.sh flake check` to ensure this PR is ready for review.
~- [ ] Added or updated unit tests.~
~- [ ] Linked a PR or a [docs tracking issue] to [document this change].~
~- [ ] Added `backport release-x.y` labels to auto-backport this PR.~

[contribution process]: https://github.com/crossplane/crossplane/tree/main/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute

## Summary by CodeRabbit

* **Chores**
  * Updated code review automation configuration
  * Removed protobuf schema validation from CI workflow
  * Updated code ownership assignments

* **Documentation**
  * Updated contributing guidelines with reference to official documentation

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -177,16 +177,18 @@ jobs:
...
+  # Protobuf schema linting
   protobuf-schemas:
     runs-on: ubuntu-24.04
 
     steps:
       - name: Checkout
         uses: actions/checkout@de0fac2e4500dabe0009e67214ff5f5447ce83dd # v6
 
-      - name: Lint and Push Protocol Buffers
+      - name: Lint Protocol Buffers
```
Comment (jbw976) [inline .github/workflows/ci.yml]: i'd argue we don't need to do the protobuf linting - we also copy protobufs from c/c to https://github.com/crossplane/function-sdk-go and we don't lint them there either - so that has precedent 🤔

File: .coderabbit.yaml [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+    - path: "**/test/**"
+      instructions: |
+        Focus on test coverage, test clarity, and proper use of testing
+        utilities. Ask about testing scenarios and edge cases. Ensure tests are
+        maintainable and cover the happy path and error conditions. Verify
+        error testing uses proper patterns (cmpopts.EquateErrors, sentinel
+        errors for complex cases).
+
+    - path: "**/design/**"
```
Comment (jbw976) [inline .coderabbit.yaml:130]: i see some patterns that are in c/c but not applicable here - want to take a pass to trim this down to only what's applicable in this repo?

**Review by jbw976 — APPROVED** [review]
a couple small comments, thanks @adamwg!

File: CODEOWNERS
Comment (tampakrap) [inline CODEOWNERS:1]: I thought it was required by the policy. In any case I'd say to keep it, it allows us also to automatically assign people on PRs and also we might want to match dirs with people in the future

Comment (jbw976) [reply CODEOWNERS:1]: are you referring to the community extension project [policies](https://github.com/crossplane/crossplane/blob/main/GOVERNANCE.md#policies-for-community-extension-projects)? 

just to be clear, the only requirement there is for OWNERS.md, there is no requirement for CODEOWNERS

File: .coderabbit.yaml (hunk 2 above)
Comment (adamwg) [reply .coderabbit.yaml:130]: `apis/` and `test/` I expect will be relevant in the near future, so I left them in. `design/` I'm less certain about but figured it was worth keeping in case we introduce design docs here as well.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (adamwg) [reply .github/workflows/ci.yml]: Yeah, makes sense. Updated.

File: CODEOWNERS
Comment (adamwg) [reply CODEOWNERS:1]: Updated the CODEOWNERS to include the additional maintainers rather than removing it completely.
